File: modules/expo-webguard/src/ExpoWebGuardModule.ts
import { NativeModule, requireNativeModule } from 'expo';

/**
 * iOS: the terminal is a WKWebView, and a touch in its content makes WebKit's internal view
 * become first responder so key events route to the page — taking the keyboard with it, with no
 * blur and no `Keyboard.dismiss` (measured, iPhone 2026-09-14, the trace behind the re-aim in
 * `keybar.tsx`). Nothing the page or React Native does can stop it, so this module refuses the
 * become at the UIKit level while the bar's field holds the keys: a tap in the terminal stays a
 * click the pane gets, and the keys stay up. Android's focused EditText survives outside touches,
 * so there the flag is a no-op.
 */
declare class ExpoWebGuardModule extends NativeModule {
  /** While true, no view inside a WKWebView may become first responder (iOS). */
  setHold(hold: boolean): Promise<void>;
}

export default requireNativeModule<ExpoWebGuardModule>('ExpoWebGuard');

File: modules/expo-webguard/android/src/main/java/expo/modules/webguard/ExpoWebGuardModule.kt
package expo.modules.webguard

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * The Android half of the responder guard — see the iOS file for why this module exists. Android's
 * focused EditText survives an outside touch and never blurs on it (measured, and the reason
 * `keybar.tsx` needs no re-aim there), so there is nothing to guard: the flag is accepted and
 * dropped.
 */
class ExpoWebGuardModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoWebGuard")

    AsyncFunction("setHold") { (_: Boolean) -> {} }
  }
}
